internal/controller/pipelinerollout_controller.go

```diff
@@ -236,3 +302,125 @@ func (r *PipelineRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return nil
 }
+
+func setPipelineDesiredStatus(obj *kubernetes.GenericObject, status string) (*kubernetes.GenericObject, error) {
+	unstruc, err := kubernetes.ObjectToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	err = unstructured.SetNestedField(unstruc.Object, status, "spec", "lifecycle", "desiredPhase")
```

Comment:
oh, nice, I didn't know about this method.

Reply:
I wonder if we should ultimately have a `GenericObject.SetNestedField()` method that this could call